VMware.CloudFoundation.PowerManagement.psm1

```diff
@@ -1494,50 +1410,43 @@ Function Get-VMsWithPowerStatus {
-                    }
-                    if ($no_of_vms.count -eq 0) {
-                        if(-not $silence) {Write-PowerManagementLogMessage -type INFO -Message "No virtual machines in the $powerstate state."}
-                    }
-                    else {
-                        $no_of_vms_string = $no_of_vms -join ","
-                        if(-not $silence) {Write-PowerManagementLogMessage -type INFO -Message "The virtual machines in the $powerstate state are: $no_of_vms_string"}
+                if (-not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "Connected to server '$server' and attempting to get the list of virtual machines..." }
+                if ($pattern) {
+                    if ($PSBoundParameters.ContainsKey('exactMatch') ) {
+                        $no_of_vms = get-vm -Server $server | Where-Object Name -EQ $pattern | Where-Object PowerState -EQ $powerstate
+                    } else {
+                        $no_of_vms = get-vm -Server $server | Where-Object Name -Match $pattern | Where-Object PowerState -EQ $powerstate
                     }
-                    Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue  -ErrorAction  SilentlyContinue | Out-Null
-                    Return $no_of_vms
-            }
-            else {
+                } else {
+                    $no_of_vms = get-vm -Server $server | Where-Object PowerState -EQ $powerstate
+                }
+                if ($no_of_vms.count -eq 0) {
+                    if (-not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "No virtual machines in the $powerstate state." }
+                } else {
+                    $no_of_vms_string = $no_of_vms -join ","
+                    if (-not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "The virtual machines in the $powerstate state are: $no_of_vms_string" }
+                }
+                Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue -ErrorAction SilentlyContinue | Out-Null
+                Return $no_of_vms
+            } else {
                 Write-PowerManagementLogMessage -Type ERROR -Message "Cannot connect to server '$server'. Check your environment and try again."
             }
-        }
-        else {
+        } else {
             Write-PowerManagementLogMessage -Type ERROR -Message "Connection to '$server' has failed. Check your environment and try again"
         }
-    }
-    Catch {
+    } Catch {
         Debug-CatchWriterForPowerManagement -object $_
-    }
-    Finally {
-        if(-not $silence) {Write-PowerManagementLogMessage -Type INFO -Message "Completed the call to the Get-VMsWithPowerStatus cmdlet."}
+    } Finally {
+        if (-not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "Completed the call to the Get-VMsWithPowerStatus cmdlet." }
```

Comment:
```powershell
        if ( -Not $silence) { Write-PowerManagementLogMessage -Type INFO -Message "Completed the call to the Get-VMsWithPowerStatus cmdlet." }
```

or

```powershell
        if (!($silence)) { Write-PowerManagementLogMessage -Type INFO -Message "Completed the call to the Get-VMsWithPowerStatus cmdlet." }
```

Reply:
I like the full notion instead of "!". Fixed all "-not" to "-Not".